### DISCUSSION TITLE
New feature?: Auto-restart (persistent) sleep timer option

### DISCUSSION THREAD

**Discussion by listenerhp** [PaulWoitaschek/Voice#1306] [Ideas]
I usually listen to audiobooks before sleep, and I use one of those cheap earphones (with a mic) and one button on them. The button works as a play/stop button. I usually set up a sleep timer to some value (10, 20 minutes), lie down, and listen to the book. Usually, this is ok, I fall asleep, playback stops after the timeout, I go back a few minutes the next day, and continue from there. If i think I'm falling asleep sooner, i just press the play/stop button on the headset to pause playback, and go to sleep.

Then there is a third option, that I don't fall asleep, and want to listen to more of the book. But(!) I don't want to turn around in bed, wake up my better half, unlock my phone, open the app, press play, click timer, listen to my better half whining about me using the phone at 3am, set timer, put phone back, miss the nightstand, have it fall down, listen to my better half whining again, and then listen for 10 more minutes, and if i'm still not asleep, repeat the above process. The other option for me in this case is just to press the play button on the earphones, and wake up four hours of bad sleep, and six chapters later, press pause, ...etc etc.

What I would really like is a feature (checkbox) in the sleep timer settings, to restart the sleep timer when I resume playing (on the headphones, it's probably the same "play" option as in the app). So if I set the sleep timer to 10 minutes, I get 10 minutes of a book, then silence. If i'm still awake, i press the headphones button (probably the same as the play button in the app), playback starts again, and automatically the sleep timer kicks in again for 10 minutes.  And if i'm still not asleep, one press, 10 more minutes. So basically, every time you start playback after the sleep timer, the sleep timer kicks in again. 

Is something like this possible, or would it be too hard to implement due to some other issues?

**Comment by geromueller** [comment]
Hey! There is even another reason: not to forget to enable the timer. Happens quite often to me and is very annoying.

**Comment by bj0hn** [comment]
I would love that feature too!!

**Comment by gartkb** [comment]
This is very needed to complete this great app.  One thing I would like to note is that i do the same but with Bluetooth. Please add! :)

**Comment by test2a** [comment]
Oh. The one consistent issue I have found is if you are listening to a book and it switches to another file being a chapter or a new track and the screen us off, whenever I open the app, it would reset to the start of the track. Bugs me every single time

**Comment by bjohas** [comment]
Would love this too, especially for bluetooth! Other words, the same function as a "shake" but with a wired or physical play/pause button. I think the problem might be that the timer is taken off when you resume play with the play button; by contrast a shake resumes play with the timer being started again.

**Comment by ObiSandwich** [comment]
Great app, but what would make it perfect is an Auto-enabled Sleep checkbox, just like the one on the excellent AntennaPod app.

**Comment by bj0hn** [comment]
In the meantime, I use the app https://play.google.com/store/apps/details?id=ch.pboos.android.SleepTimer&gl=US

The app does exactly what I needed for all player app.
